drivers/input/touchscreen_upper.c

```diff
@@ -90,7 +92,7 @@ static const struct file_operations g_touch_fops =
   touch_open,     /* open */
   touch_close,    /* close */
   touch_read,     /* read */
-  NULL,           /* write */
+  touch_write,    /* write */
```

Comment:
Shouldn't it depend on some menuconfig option? A common/real touchscreen device should have a write operation by default. Talking about sending data to/over touchscreen, how we could add support to HotKnot ?

Reply:
Config has been added, and different macros have been added in touchscreen and buttons respectively for selection

Reply:
@acassis the callback just add a dummy function with several bytes, does it make sense to couple the general code with uinput?

Reply:
@xiaoxiang781216 good question, at least now it will not be coupled case the option is not enabled in the menu. So user developing standard touchscreen drivers will "not see" the write function.